### PR TITLE
Add clang static analyzer ci (#187)

### LINKED ISSUE #187: Static analysis or linting tools for the project (and CI)
**Is your feature request related to a problem? Please describe.**
For now, the only code analysis tool that analyzes this repository is codeQL (Where can I see the report?). But there are many alternatives such as the clang static analyzer, clang-tidy(it's a linter but it could be interesting).

**Describe the solution you'd like**
Add the clang static analyzer in the CI

```
meson build
scan-build meson compile -C build
scan-now <report>
```

**Describe alternatives you've considered**
Use other alternatives such as clang-tidy (there is a way to only lint changed file)

https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py

```
meson build
meson compile -C build
cd build && run-clang-tidy.py -header-filter ".*"
```
[log.zip](https://github.com/rizinorg/rizin/files/5691755/log.zip)
(if std=c11)

**Additional context**
I think that the more measures there are, the better we can assess the state of the project.

We need more than one tool.

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**
A new workflow "Code analyzis (clang)"
Use clang static analyzer and push the result (sarif format)

**Test plan**
Check the CI result

**Closing issues**
closes #187

### CODE REVIEW

File: .github/workflows/clang-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: "Code analyzis (clang) - action"
+
+on:
+  push:
+    branches:
+    - 'master'
```
Comment (ret2libc) [inline .github/workflows/clang-analysis.yml]: no need for this

File: .github/workflows/clang-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - 'dev'
+    - 'stable'
+  pull_request:
+    branches:
+    - master
```
Comment (ret2libc) [inline .github/workflows/clang-analysis.yml]: nor for this.

File: .github/workflows/clang-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Analysis
+      run: |
+        export PATH=${HOME}/Library/Python/3.8/bin:${HOME}/Library/Python/3.9/bin:${HOME}/.local/bin:${PATH}
+        scan-build -sarif -o reports "meson build && ninja -C build"
+
+    - name: Move reports
+      run: mv reports/*/*.sarif reports/
+
+    - name: Upload SARIF file
+      uses: github/codeql-action/upload-sarif@v1
```
Comment (ret2libc) [inline .github/workflows/clang-analysis.yml:52]: where is it possible to see the result?

Comment (ghost) [reply .github/workflows/clang-analysis.yml:52]: Location:
simplescreenrecorder-2020-12-16_22 26 40

TLDR: The report is not available for this build because it was not run on a base branch.

2020-12-16-222551_651x314_scrot

File: .github/workflows/clang-analysis.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/clang-analysis.yml]: I removed master

File: .github/workflows/clang-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name: "Code analyzis (clang)"
```
Comment (yossizap) [inline .github/workflows/clang-analysis.yml]: analysis*

**Review by XVilka — CHANGES_REQUESTED** [review]
Amazing. But Ubuntu uses old Clang (9.0) while latest (11.0) offers better analysis. What about using official LLVM from https://apt.llvm.org/ ?

See https://repology.org/project/llvm/versions

**Review by wargio — APPROVED** [review]
Love it, but yeah. better use the official LLVM repo.

**Comment by ghost** [comment]
> Amazing. But Ubuntu uses old Clang (9.0) while latest (11.0) offers better analysis. What about using official LLVM from https://apt.llvm.org/ ?

I switched to clang-11

**Review by ret2libc — APPROVED** [review]

**Comment by ghost** [comment]
I can't merge because 'XVilka requested changes'.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Thank you! Lets see how much of warnings it will produce :)
